core/src/main/java/tc/oc/pgm/friends/FriendRegistry.java

```diff
@@ -0,0 +1,20 @@
+package tc.oc.pgm.friends;
+
+import org.bukkit.entity.Player;
+import org.bukkit.event.Listener;
+import tc.oc.pgm.util.friends.FriendProvider;
+
+/**
+ * The FriendRegistry is responsible for managing a {@link FriendProvider} to consistently supply
+ * whether two players are friends
+ */
+public interface FriendRegistry extends Listener, FriendProvider {
```

Comment:
Seems weird to have these friend things in PGM while not being able to interact with it without having a separate
plugin. All things useable with the friend feature seems to care if to parties is related in a certain way. Why not have the settings have an option to add custom relation checks in a way. This way other plugins could use their own groups as well (based on ranks, teams, staff...)

Reply:
I thought a little about this, and I think the best way would be to have a way to register some `PartyRelation`.

This way whenever the settings that could care about parties is queried they can look for any custom `PartyRelation`s in addition to the hardcoded options.

I believe this solves the support for friends, while not locking the possibility to only friends.

Reply:
@KingOfSquares 
I like your thinking on this topic. I was initially unsure when implementing this how specific to get. 

I think the settings area of PGM is in dire need of a comprehensive update, cleanup to the commands and perhaps less reliance on hard coded values. That was not something I was planning to include in this PR, but could very be a future project.

I’d like some other feedback on whether switching everything to `PartyRelation` would be the right way to go. If you had any ideas in regards to the specifics though, let me know! 

Atm, I’m not sure what other sort of groups would use the same features as friends though? It may be confusing if more than one plugin utilized the same formatting of italicizing friend names, I feel making it specific reserves that formatting consistency. But I understand the desire to make it more generic. 

Thanks again for the feedback! 😄